### ISSUE TITLE
[masthead v2]: merge into `feat/carbon-for-ibm-dotcom-v2`

### ISSUE THREAD

**Issue by kennylam** [carbon-design-system/carbon-for-ibm-dotcom#10675]
JIRA: https://jsw.ibm.com/browse/ADCMS-3628

This issue tracks the work for merging `masthead-v2` into the `feat/carbon-for-ibm-dotcom-v2` feature branch. Once the merge is complete, work can begin updating the masthead to Carbon v11, as well as updating to use `@carbon/web-components` v2.

```[tasklist]
### Tasks
- [ ] https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/10872
- [x] [masthead-v2]: upgrade to Carbon v11
- [x] [masthead-v2]: upgrade components to use `@carbon/web-components` v2
```

**Comment by andy-blum** [comment]
@jkaeser @kennylam is this work complete now?

**Comment by jkaeser** [comment]
@andy-blum Yes, this was completed once https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/10952 was merged. Closing this epic out.
